### PR TITLE
initial 1.86.0 version

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
mcl 1.86.0 for ❄️

- dependency for mclbn256
- [upstream](https://github.com/herumi/mcl/blob/v1.86.0/)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - clang                                                 # [linux and aarch64]
+    - clangxx                                               # [linux and aarch64]
+    - cmake
+    - make                                                  # [unix]
+    - ninja                                                 # [win]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - ninja-base                                                 # [win]
```
`ninja` includes `python` but it seems you don't need it

File: recipe/bld.bat [hunk 2]
```diff
@@ -0,0 +1,18 @@
+mkdir build
```
Comment (skupr-anaconda) [inline recipe/bld.bat]:
```suggestion
:: cmd
echo "Building %PKG_NAME%."

mkdir build
```

File: recipe/bld.bat [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+cd build
+if %ERRORLEVEL% neq 0 exit 1
+
+cmake -GNinja ^
+    -DCMAKE_PREFIX_PATH="%LIBRARY_PREFIX%" ^
+    ..
```
Comment (skupr-anaconda) [inline recipe/bld.bat]:
```suggestion
:: Generate the build files.
echo "Generating the build files..."
cmake .. %CMAKE_ARGS% ^
      -G"Ninja" ^
      -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
      -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
      -DCMAKE_BUILD_TYPE=Release
```

File: recipe/build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+mkdir build
+cd build
+
+if [[ $target_platform == linux-aarch64 ]]; then
+    cmake   -DCMAKE_INSTALL_PREFIX=$PREFIX \
+            -DCMAKE_CXX_COMPILER=clang++ \
+    ..
```
Comment (skupr-anaconda) [inline recipe/build.sh]:
```suggestion
    cmake .. ${CMAKE_ARGS} \
          -GNinja \
          -DCMAKE_PREFIX_PATH=${PREFIX} \
          -DCMAKE_INSTALL_PREFIX=${PREFIX} \
          -DCMAKE_BUILD_TYPE=Release \
          -DCMAKE_CXX_COMPILER=clang++
```

File: recipe/build.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+else
+    cmake   -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    ..
```
Comment (skupr-anaconda) [inline recipe/build.sh]:
```suggestion
    cmake .. ${CMAKE_ARGS} \
          -GNinja \
          -DCMAKE_PREFIX_PATH=${PREFIX} \
          -DCMAKE_INSTALL_PREFIX=${PREFIX} \
          -DCMAKE_BUILD_TYPE=Release \
```

File: recipe/bld.bat (hunk 3 above)
Comment (skupr-anaconda) [reply recipe/bld.bat]: It's from the template https://github.com/anaconda-distribution/distro-incubator/tree/main/skel

File: recipe/build.sh (hunk 4 above)
Comment (JeanChristopheMorinPerso) [reply recipe/build.sh:9]: We don't need to set `CMAKE_CXX_COMPILER`.

File: recipe/build.sh [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+mkdir build
+cd build
+
+if [[ $target_platform == linux-aarch64 ]]; then
+    cmake   -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (JeanChristopheMorinPerso) [inline recipe/build.sh]: We should be explicit about the build type (Release) and we should also pass `$CMAKE_ARGS`.

File: recipe/build.sh (hunk 4 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/build.sh]: the -DCMAKE_PREFIX_PATH=${PREFIX} 
giving me such error:

CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/home/mwaszkiewicz/miniconda3/conda-bld/mcl_1702977335269/work/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/build.sh [hunk 7]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+mkdir build
+cd build
+
+if [[ $target_platform == linux-aarch64 ]]; then
+    cmake   ${CMAKE_ARGS} \
+            -DCMAKE_INSTALL_PREFIX=$PREFIX \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_CXX_COMPILER=clang++ \
```
Comment (boldorider4) [inline recipe/build.sh:10]: why does this need to be specified on `linux-aarch64`? doesn't it use the same toolchain as other unix platforms?

File: recipe/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+build:
+  skip: True  # [s390x]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - clang    # [linux and aarch64]
```
Comment (boldorider4) [inline recipe/meta.yaml:24]: I see this is the same as the `build.sh`. Can you comment why this is necessary?

**Review by boldorider4 — APPROVED** [review]
LGTM but maybe add couple comments where noted.

File: recipe/meta.yaml (hunk 8 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:24]: yes, it is clang on linux aarch64, see 
https://github.com/herumi/mcl/blob/79774aa88da73d58d5fd4da209cc5373770f8009/CMakeLists.txt#L150-L166

File: recipe/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    - test -f $PREFIX/lib/libmclbn384_256${SHLIB_EXT}       # [unix]
+    - test -f $PREFIX/include/mcl/bn256.hpp                 # [unix]
+    - test -f $PREFIX/include/mcl/bn_c256.h                 # [unix]
+    - test -f $PREFIX/include/mcl/bn.hpp                    # [unix]
+    - test -f $PREFIX/include/mcl/bn.h                      # [unix]
+
+    - if not exist %LIBRARY_BIN%\mcl.dll          exit 1     # [win]
+    - if not exist %LIBRARY_BIN%\mclbn256.dll     exit 1     # [win]
+    - if not exist %LIBRARY_BIN%\mclbn384.dll     exit 1     # [win]
+    - if not exist %LIBRARY_BIN%\mclbn384_256.dll exit 1     # [win]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:48]:
```suggestion
    - if not exist %LIBRARY_BIN%\mclbn384_256.dll exit 1     # [win]
    - if not exist %LIBRARY_LIB%\mcl.lib exit 1 # [win]
    - if not exist %LIBRARY_LIB%\mclbn256.lib exit 1 # [win]
    - if not exist %LIBRARY_LIB%\mclbn384.lib exit 1 # [win]
    - if not exist %LIBRARY_LIB%\mclbn384_256.lib exit 1 # [win]
```

File: recipe/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    - if not exist %LIBRARY_INC%\mcl\bn256.hpp    exit 1     # [win]
+    - if not exist %LIBRARY_INC%\mcl\bn_c256.h    exit 1     # [win]
+    - if not exist %LIBRARY_INC%\mcl\bn256.hpp    exit 1     # [win]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 8 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:24]: Please add a comment in the recipe.

File: recipe/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - clang    # [linux and aarch64]
+    - clangxx  # [linux and aarch64]
+    - cmake
+    - make  # [unix]
+    - patch  # [not win]
+    - m2-patch  # [win]
+  host:
+    - gmp  # [unix]
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:32]:
```suggestion
    - gmp  # [unix]
  run:
    # ping through gmp run_exports
    - gmp  # [unix]
```

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Thanks @M-Waszkiewicz-Anaconda !

**Pull request merged**